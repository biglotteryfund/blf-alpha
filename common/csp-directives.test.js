// @ts-nocheck
/* eslint-env jest */
'use strict';

const cspDirectives = require('./csp-directives');

describe('cspDirectives', () => {
    test('should return default csp directives', () => {
        expect(cspDirectives()).toMatchSnapshot();
    });

    test('should return additional directives when hotjar is enabled', () => {
        expect(cspDirectives({ enableHotjar: true })).toMatchSnapshot();
    });

    test('should return additional directives when localhost is enabled', () => {
        expect(cspDirectives({ allowLocalhost: true })).toMatchSnapshot();
    });
});
