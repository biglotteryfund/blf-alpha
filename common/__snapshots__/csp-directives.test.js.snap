// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cspDirectives should return additional directives when hotjar is enabled 1`] = `
Object {
  "baseUri": Array [
    "'self'",
  ],
  "childSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "www.google.com",
    "https://vars.hotjar.com",
  ],
  "connectSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://*.hotjar.com:*",
    "https://*.hotjar.com:*",
    "https://vc.hotjar.io:*",
    "wss://*.hotjar.com",
  ],
  "defaultSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
  ],
  "fontSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "data:",
    "use.typekit.net",
    "http://static.hotjar.com",
    "https://static.hotjar.com",
  ],
  "frameSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "https://vars.hotjar.com",
  ],
  "imgSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "data:",
    "localhost",
    "stats.g.doubleclick.net",
    "via.placeholder.com",
    "biglotteryfund-assets.imgix.net",
    "https://insights.hotjar.com",
    "http://static.hotjar.com",
    "https://static.hotjar.com",
  ],
  "reportUri": "https://sentry.io/api/226416/csp-report/?sentry_key=53aa5923a25c43cd9a645d9207ae5b6c",
  "scriptSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "'unsafe-eval'",
    "'unsafe-inline'",
    "http://static.hotjar.com",
    "https://static.hotjar.com",
    "https://script.hotjar.com",
    "'unsafe-eval'",
    "'unsafe-inline'",
  ],
  "styleSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "'unsafe-inline'",
    "*.typekit.net",
  ],
}
`;

exports[`cspDirectives should return additional directives when localhost is enabled 1`] = `
Object {
  "baseUri": Array [
    "'self'",
  ],
  "childSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://127.0.0.1:*",
    "http://localhost:*",
    "ws://localhost:*",
    "www.google.com",
  ],
  "connectSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://127.0.0.1:*",
    "http://localhost:*",
    "ws://localhost:*",
  ],
  "defaultSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://127.0.0.1:*",
    "http://localhost:*",
    "ws://localhost:*",
  ],
  "fontSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://127.0.0.1:*",
    "http://localhost:*",
    "ws://localhost:*",
    "data:",
    "use.typekit.net",
  ],
  "frameSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://127.0.0.1:*",
    "http://localhost:*",
    "ws://localhost:*",
  ],
  "imgSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://127.0.0.1:*",
    "http://localhost:*",
    "ws://localhost:*",
    "data:",
    "localhost",
    "stats.g.doubleclick.net",
    "via.placeholder.com",
    "biglotteryfund-assets.imgix.net",
  ],
  "reportUri": "https://sentry.io/api/226416/csp-report/?sentry_key=53aa5923a25c43cd9a645d9207ae5b6c",
  "scriptSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://127.0.0.1:*",
    "http://localhost:*",
    "ws://localhost:*",
    "'unsafe-eval'",
    "'unsafe-inline'",
  ],
  "styleSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "http://127.0.0.1:*",
    "http://localhost:*",
    "ws://localhost:*",
    "'unsafe-inline'",
    "*.typekit.net",
  ],
}
`;

exports[`cspDirectives should return default csp directives 1`] = `
Object {
  "baseUri": Array [
    "'self'",
  ],
  "childSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "www.google.com",
  ],
  "connectSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
  ],
  "defaultSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
  ],
  "fontSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "data:",
    "use.typekit.net",
  ],
  "frameSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
  ],
  "imgSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "data:",
    "localhost",
    "stats.g.doubleclick.net",
    "via.placeholder.com",
    "biglotteryfund-assets.imgix.net",
  ],
  "reportUri": "https://sentry.io/api/226416/csp-report/?sentry_key=53aa5923a25c43cd9a645d9207ae5b6c",
  "scriptSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "'unsafe-eval'",
    "'unsafe-inline'",
  ],
  "styleSrc": Array [
    "'self'",
    "*.biglotteryfund.org.uk",
    "*.tnlcommunityfund.org.uk",
    "*.google.com",
    "*.facebook.com",
    "*.twitter.com",
    "*.gstatic.com",
    "*.twimg.com",
    "*.youtube.com",
    "*.vimeo.com",
    "cdn.polyfill.io",
    "cdn.syndication.twimg.com",
    "maxcdn.bootstrapcdn.com",
    "ajax.googleapis.com",
    "cdnjs.cloudflare.com",
    "platform.twitter.com",
    "sentry.io",
    "syndication.twitter.com",
    "www.google-analytics.com",
    "use.typekit.net",
    "cms.blf.digital",
    "'unsafe-inline'",
    "*.typekit.net",
  ],
}
`;
